src-tauri/src/main.rs

```diff
@@ -0,0 +1,308 @@
+// Prevents additional console window on Windows in release, DO NOT REMOVE!!
+#![cfg_attr(all(not(debug_assertions), target_os = "windows"), windows_subsystem = "windows")]
+
+mod state;
+
+use log::info;
+use state::AppState;
+use tauri::{Manager, State, RunEvent, AppHandle};
+use tauri_plugin_shell::{ShellExt, process::CommandEvent};
+
+#[tauri::command]
+fn greet(name: &str) -> String {
+    format!("Hello, {}! You've been greeted from Rust!", name)
+}
+
+#[tauri::command]
+async fn is_phosphobot_server_running() -> bool {
+    let client = tauri_plugin_http::reqwest::Client::new();
+    match client.get("http://127.0.0.1:8432/status").send().await {
+        Ok(resp) => {
+            log::info!("Status check response: {:?}", resp.status());
+            resp.status().is_success()
+        },
+        Err(e) => {
+            log::warn!("Status check failed: {}", e);
+            false
+        }
+    }
+}
+
+#[tauri::command] 
+async fn start_phosphobot_server<'a>(
+    app: tauri::AppHandle,
+    state: State<'a, AppState>,
+) -> Result<String, String> {
+    info!("Attempting to launch sidecar: phosphobot");
+    
+    let sidecar_command = app.shell().sidecar("phosphobot").map_err(|e| {
+        log::error!("Failed to get sidecar: {}", e);
+        e.to_string()
+    })?;
+    
+    log::info!("Sidecar command successfully created.");
+    
+    let (mut rx, child) = sidecar_command
+        .args([
+            "run",
+            "--host=127.0.0.1", 
+            "--port=8432",
+            "--simulation=headless",
+            "--simulate-cameras",
+            "--only-simulation",
```

Comment:
please remove --simulation-headless and --simulate-cameras as they make the app unable to connect to real cameras and to real robots